src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c

```diff
@@ -28,44 +46,498 @@ const ActorInit Dm_Char01_InitVars = {
     (ActorFunc)DmChar01_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AAAAB0[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 300, ICHAIN_STOP),
 };
 
-#endif
+static s16 D_80AAAAB4 = 0;
+
+void DmChar01_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DmChar01* this = THIS;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    Actor_SetScale(&this->dyna.actor, 1.0f);
+
+    this->unk_346 = 0;
+    this->unk_34D = 0;
+    D_80AAAE24 = 0;
+    D_80AAAE26 = 0;
+
+    switch (this->dyna.actor.params) {
+        case 0:
+            if (gSaveContext.weekEventReg[20] & 2) {
+                this->unk_34C = 2;
+                this->actionFunc = func_80AA8F1C;
+                break;
+            }
+
+            if (gSaveContext.sceneSetupIndex == 0) {
+                globalCtx->envCtx.unk_1F = 5;
+                globalCtx->envCtx.unk_20 = 5;
+            }
+            this->unk_348 = 255.0f;
+
+            for (i = 0; i < ARRAY_COUNT(this->unk_1AC); i++) {
+                this->unk_1AC[i] = ovl_dm_char01_Vtx_1BE0[i].v.ob[1] * 409.6f;
+            }
+
+            DynaPolyActor_Init(&this->dyna, 0);
+            DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_mtoride_Colheader_009E4C);
+
+            this->unk_34D = 1;
```

Comment:
```suggestion
            this->unk_34D = true;
```
This is also boolean and can be changed